### DISCUSSION TITLE
Thoughts on porting Fabric to Python?

### DISCUSSION THREAD

**Discussion by andreamancuso** [idom-team/idom#1246] [Ideas]
Hello!
First of all, let me tell you how impressed I am to see what you guys accomplished with ReactPy. Seriously ambitious and amazing.

I understand based on [this answer](https://github.com/reactive-python/reactpy/discussions/1177) that you've a bigger fish to fry at the moment, however I still would like to ask: what if somebody wanted to port https://github.com/facebook/react-native/blob/main/packages/react-native/Libraries/Renderer/implementations/ReactFabric-prod.js to Python (primarily this file, not the entire React Native ecosystem), where/how would you start?
Have you written a reconciler algorithm?

Cheers!

**Comment by Archmonger** [comment]
I haven't read the react fabric source code. However, implementation of anything React-Native related is going to need tinkering with the [ReactPy Client](https://github.com/reactive-python/reactpy/tree/e09d09c234dbcbdb045ee12077d8ac47cb6729a0/src/js/packages/%40reactpy/client/src).

I don't see a world where anything React-Native related could be server-side rendered, so client-side components would need to get ported to ReactPy core first. Currently, client-side components only [exist in Django](https://reactive-python.github.io/reactpy-django/latest/reference/template-tag/#pyscript-component) due to convenience (specifically due to Django having built-in template tags and settings framework).

In summary, to get React Native running within ReactPy the "order of operations" would be the following:
1) https://github.com/reactive-python/reactpy/issues/653
2) https://github.com/reactive-python/reactpy/issues/110
3) https://github.com/reactive-python/reactpy/issues/570

**Reply by Archmonger** [reply]
As a note about the template tags, someone has an [open PR](https://github.com/reactive-python/reactpy-django/pull/199) in `reactpy-django` that is functional and utilizes `Jinja` template tags. This is convenient because `Jinja` is exactly what ReactPy core will eventually use for template tags. If I recall, the merge blocker for that PR was a lack of tests. It is likely very easy to shove that source code into ReactPy core and add some tests.

Then regarding client-side components, it would also be an exercise in _mostly_ copy-pasting the following from ReactPy-Django: [1](https://github.com/reactive-python/reactpy-django/blob/c5b5c81854c2dd0e59f325a0a708ae664c768c76/src/reactpy_django/templates/reactpy/pyscript_component.html), [2](https://github.com/reactive-python/reactpy-django/blob/c5b5c81854c2dd0e59f325a0a708ae664c768c76/src/reactpy_django/templates/reactpy/pyscript_setup.html), [3](https://github.com/reactive-python/reactpy-django/blob/c5b5c81854c2dd0e59f325a0a708ae664c768c76/src/reactpy_django/components.py#L261-L284), [4](https://github.com/reactive-python/reactpy-django/blob/c5b5c81854c2dd0e59f325a0a708ae664c768c76/src/reactpy_django/pyscript/layout_handler.py), [5](https://github.com/reactive-python/reactpy-django/blob/c5b5c81854c2dd0e59f325a0a708ae664c768c76/src/reactpy_django/pyscript/component_template.py)
